tests/unit/test_ldap_backend.py

```diff
@@ -194,35 +250,71 @@ def test_search_valid_username_invalid_group(self):
         username = 'john_connor'
         password = 'HastaLavista'
         user_dn = 'uid={},ou=users,dc=example,dc=com'.format(username)
-        mock_user_res = (user_dn, LDAPAuthenticationBackendTestCase.directory[user_dn])
+        mock_user_res_id = 1234
+        mock_user_res = (ldap.RES_SEARCH_RESULT, [(user_dn, LDAPAuthenticationBackendTestCase.directory[user_dn])])
 
         groupname = 'invalid_group'
         group_dn = 'cn={groupname},ou=groups,dc=example,dc=com'.format(groupname=groupname)
-        mock_group_res = []
+        mock_group_res_id = 9999
+        mock_group_res = (ldap.RES_SEARCH_RESULT, None)
 
         user = {"base_dn": "ou=users,dc=example,dc=com", "search_filter": "(uid={username})", "scope": "onelevel"}
         group = {"base_dn": "ou=groups,dc=example,dc=com", "search_filter": "(&(cn=%s)(memberUid={username}))"%groupname, "scope": "subtree"}
 
-        self.ldapobj.search_s.seed(user["base_dn"], ldap.SCOPE_ONELEVEL, user["search_filter"].format(username=username))([mock_user_res])
-        self.ldapobj.search_s.seed(group["base_dn"], ldap.SCOPE_SUBTREE, group["search_filter"].format(username=username))(mock_group_res)
+        self.ldapobj._search.seed(user["base_dn"], ldap.SCOPE_ONELEVEL, user["search_filter"].format(username=username))(mock_user_res_id)
+        self.ldapobj._search.seed(group["base_dn"], ldap.SCOPE_SUBTREE, group["search_filter"].format(username=username))(mock_group_res_id)
+        self.ldapobj._result.seed(mock_user_res_id, all=0)(mock_user_res)
+        self.ldapobj._result.seed(mock_group_res_id, all=0)(mock_group_res)
 
         result = _do_simple_bind('ldap://fakeldap.example.com/', 'cn=manager,dc=example,dc=com', 'ldaptest', user_search=user, group_search=group, username=username, password=password)
 
         self.assertEquals(self.ldapobj.methods_called(), [
                 'initialize',
                 'simple_bind_s',
                 'whoami_s',
-                'search_s',
+                'search',
+                'result',
+                'result',
                 'initialize',
                 'simple_bind_s',
                 'whoami_s',
                 'unbind',
-                'search_s',
+                'search',
+                'result',
                 'unbind'
             ]
         )
         self.assertFalse(result)
 
+    def test_search_with_reference_result(self):
+        # This is for returning the referral object at calling 'result' method of LDAPObject
+        self.mock_referral = [
+            (None, ['ldap://fakeldap2.example.com/ou=cyberdyne,dc=example,dc=com']),
+        ]
+
+        user = {
+            "base_dn": "ou=users,dc=example,dc=com",
+            "search_filter": "(uid={username})",
+            "scope": "subtree",
+        }
+
+        result = _do_simple_bind('ldap://fakeldap.example.com/', '', '',
```

Comment:
The name is fine. I just saw many references to the string `ldap://fakeldap.example.com/` so to avoid mistyping/confusion it might be easier to set this as a var:

```
LDAP_URL = 'ldap://fakeldap.example.com/'
...
result = _do_simple_bind(LDAP_URL, '', '',
...
```

But again, this isn't a blocker, just something I noticed and had been there prior to this PR.